### PR TITLE
chore: added support for avatar hashing function

### PR DESCRIPTION
**Pull request by ayadav-eightfold**
chore: added support for avatar hashing function

### CODE REVIEW

**Review by pbahukhandi-eightfold — APPROVED** [review]

File: src/components/Avatar/Avatar.tsx [hunk 1]
```diff
@@ -18,11 +18,16 @@ export const AVATAR_CLASS_SET = [
...
-            () => Math.floor(Math.random() * 100) % AVATAR_CLASS_SET.length,
-            []
-        );
+    (
+        { children, classNames, style, hashingFunction },
+        ref: Ref<HTMLDivElement>
+    ) => {
+        const colorSetIndex = useMemo(() => {
+            if (hashingFunction) {
+                return hashingFunction();
```
Comment (pbahukhandi-eightfold) [inline src/components/Avatar/Avatar.tsx:27]: @ychhabra-eightfold @dkilgore-eightfold : to review

**Review by dkilgore-eightfold — APPROVED** [review]

File: src/components/Avatar/Avatar.tsx [hunk 2]
```diff
@@ -18,11 +18,16 @@ export const AVATAR_CLASS_SET = [
 ];
```
Comment (ayadav-eightfold) [inline src/components/Avatar/Avatar.tsx:18]: expose theme set as "theme": type : OcThemeNames(src/components/ConfigProvider/Theming/Theming.types.ts)
prop randomiseTheme: if true, use Math.floor(Math.random() * 100) % AVATAR_CLASS_SET.length;
passing the hashing function as prop

File: src/components/Avatar/Avatar.tsx [hunk 3]
```diff
@@ -65,6 +76,9 @@ export const Avatar: FC<AvatarProps> = React.forwardRef(
             fontSize = '18px',
             iconProps,
             children,
+            hashingFunction,
+            theme,
```
Comment (ychhabra-eightfold) [inline src/components/Avatar/Avatar.tsx:84]: can we make default blue

File: src/components/Avatar/Avatar.tsx [hunk 4]
```diff
@@ -18,16 +19,26 @@ export const AVATAR_CLASS_SET = [
...
+                return Math.floor(hashingFunction()) % AVATAR_THEME_SET.length;
+            }
+            return 0;
+        }, []);
 
         const avatarClasses: string = mergeClasses([
             styles.wrapperStyle,
             classNames,
-            AVATAR_CLASS_SET[colorSetIndex],
+            styles[theme] || AVATAR_THEME_SET[colorSetIndex],
```
Comment (ychhabra-eightfold) [inline src/components/Avatar/Avatar.tsx]: dont think this will work `styles[theme]` can you confirm ?

File: src/components/Avatar/Avatar.tsx [hunk 5]
```diff
@@ -5,33 +5,48 @@ import styles from './avatar.module.scss';
...
+
         return (
-            <div ref={ref} className={avatarClasses} style={style}>
+            <div
+                ref={ref}
+                className={avatarClasses}
+                style={{
+                    backgroundColor: `var(--${theme}-color-20)`,
+                    color: `var(--${theme}-color)`,
+                    border: `solid 1px var(--${theme}-color-30)`,
```
Comment (ychhabra-eightfold) [inline src/components/Avatar/Avatar.tsx:46]: can we just create a theme to stype map and use that instead of this

**Review by ychhabra-eightfold — CHANGES_REQUESTED** [review]

File: src/components/Avatar/Avatar.tsx (hunk 3 above)
Comment (ayadav-eightfold) [reply src/components/Avatar/Avatar.tsx:84]: If we keep a default theme, we'll never be able to use the hashing function on the basis of priority order we had decided

**Pull request closed without merging**